### PR TITLE
soc: psoc6: update pinctrl for PSoC6 MCU (legacy)

### LINKED ISSUE #52434: drivers: pinctrl: Use common pinctrl DTS bindings for Cypress psoc6
**Is your enhancement proposal related to a problem? Please describe.**
As noted in #39740 the Cypress PSoC6 implementation uses a custom devicetree pinctrl representation (as seen in `dts/arm/cypress/psoc6-pinctrl.dtsi`).

**Describe the solution you'd like**
The Cypress PSoC6 implementation should use one of the standard Zephyr pinctrl devicetree representations as described in https://docs.zephyrproject.org/latest/hardware/pinctrl/index.html#devicetree-representation along with a pinctrl driver.

**Describe alternatives you've considered**
None.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by npal-cy**
update pinctrl for PSoC6 MCU (legacy)

Fixes: https://github.com/zephyrproject-rtos/zephyr/issues/52434

### CODE REVIEW

**Review by ifyall — DISMISSED** [review]

**Review by decsny — DISMISSED** [review]

File: boards/cypress/cy8ckit_062_ble/cy8ckit_062_ble_cy8c6347_m0_defconfig [hunk 1]
```diff
@@ -11,3 +11,6 @@ CONFIG_BUILD_OUTPUT_HEX=y
 
 # UART driver
 CONFIG_SERIAL=y
+
+# Pinctrl driver
+CONFIG_PINCTRL=y
```
Comment (gmarull) [inline boards/cypress/cy8ckit_062_ble/cy8ckit_062_ble_cy8c6347_m0_defconfig:16]: please do not enable pinctrl at board level, make drivers select the option

File: boards/cypress/cy8ckit_062_ble/cy8ckit_062_ble_cy8c6347_pinctrl.dtsi [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2023 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ * SPDX-License-Identifier: Apache-2.0
```
Comment (gmarull) [inline boards/cypress/cy8ckit_062_ble/cy8ckit_062_ble_cy8c6347_pinctrl.dtsi:4]: nit: files are usually named `-pinctrl.dtsi`, not `_pinctrl.dtsi`

File: soc/infineon/cat1a/psoc6_legacy/pinctrl_soc.h [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+ * Bit definition in PINMUX field
+ */
+#define SOC_PINMUX_PORT_POS             (0)
+#define SOC_PINMUX_PORT_MASK            (0xFFul << SOC_PINMUX_PORT_POS)
+#define SOC_PINMUX_PIN_POS              (8)
+#define SOC_PINMUX_PIN_MASK             (0xFFul << SOC_PINMUX_PIN_POS)
+#define SOC_PINMUX_HSIOM_FUNC_POS       (16)
+#define SOC_PINMUX_HSIOM_MASK           (0xFFul << SOC_PINMUX_HSIOM_FUNC_POS)
+#define SOC_PINMUX_SIGNAL_POS           (24)
+#define SOC_PINMUX_SIGNAL_MASK          (0xFFul << SOC_PINMUX_SIGNAL_POS)
```
Comment (gmarull) [inline soc/infineon/cat1a/psoc6_legacy/pinctrl_soc.h:37]: nit: ise GENMASK, BIT, etc. macros

**Review by gmarull — CHANGES_REQUESTED** [review]

File: boards/cypress/cy8ckit_062_ble/cy8ckit_062_ble_cy8c6347_m0_defconfig (hunk 1 above)
Comment (npal-cy) [reply boards/cypress/cy8ckit_062_ble/cy8ckit_062_ble_cy8c6347_m0_defconfig]: updated

**Review by gmarull — APPROVED** [review]

**Review by ifyall — APPROVED** [review]

**Pull request merged**